src/emqtt.erl

```diff
@@ -1559,7 +1567,14 @@ handle_event(info, {Closed, _Sock}, connected, State) when ?SOCK_CLOSED(Closed)
 
 handle_event(info, {Closed, _Sock}, waiting_for_connack, State) when ?SOCK_CLOSED(Closed) ->
     ?LOG(info, "socket_closed_before_connack", #{}, State),
-    maybe_reconnect(Closed, State);
+    #state{socket = Via} = State,
+    case take_call(call_id(connect, Via), State) of
+        {value, #call{from = From}, _NewState} ->
+            Error = {error, Closed},
+            shutdown_reply(Closed, From, Error);
```

Comment:
since we have just added reconnect handler, maybe we don't need this change anymore? 

I think the `reconnect` option is really mean to keep the client *online* even the socket is closed before connack is received and then the next reconnect may end up to another broker node where the client is welcomed.  

The user/caller of this process could kill it if it cannot wait due to timeout or whatever reason.

WDYT?

Reply:
I added this because it made the MQTT bridge hang for a long while starting the client pool.  :thinking: 

I was thinking that the reconnect option would "take effect" after the client connect once.

Also, if that's the case, then isn't this clause inconsistent with this behavior of never shutting down?

https://github.com/emqx/emqtt/blob/ea338854dce9f74dca01fd74b080b43cae2713f5/src/emqtt.erl#L1153-L1163

Here, if there is a `connect` call ongoing, the process terminates, which is basically what the change here does.  :thinking: